File: app/src/main/java/com/dgd/earthquakes/common/HostedFragment.java
package com.dgd.earthquakes.common;

import android.app.Activity;
import android.support.v4.app.Fragment;

/**
 * Created by Dan on 19/07/2017.
 */

public abstract class HostedFragment<T> extends Fragment implements IHostedFragment {
    protected T host;
    private boolean hasProgress;

    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);
        host = (T) activity;
    }

    @Override
    public void onDetach() {
        super.onDetach();
        host = null;
    }

    protected void setHasProgress(boolean hasProgress) {
        this.hasProgress = hasProgress;
    }

    @Override
    public boolean hasProgress() {
        return hasProgress;
    }

    @Override
    public void showProgress() {}

    @Override
    public void hideProgress() {}
}

File: app/src/main/java/com/dgd/earthquakes/common/HostedFragment.kt
package com.dgd.earthquakes.common

import android.content.Context
import android.support.v4.app.Fragment

/**
 * Created by Dan
 * on 19/07/2017.
 */

abstract class HostedFragment<T : FragmentContract.IHost> : Fragment(), FragmentContract.IFragment {
    protected var host: T? = null
    private var hasProgress: Boolean = false

    override fun onAttach(activity: Context?) {
        super.onAttach(activity)
        host = activity as T?
    }

    override fun onDetach() {
        super.onDetach()
        host = null
    }

    protected fun setHasProgress(hasProgress: Boolean) {
        this.hasProgress = hasProgress
    }

    override fun hasProgress(): Boolean {
        return hasProgress
    }

    override fun showProgress() {}

    override fun hideProgress() {}
}
